hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/FlinkOptions.java

```diff
@@ -343,12 +343,25 @@ private FlinkOptions() {
       .noDefaultValue()
       .withDescription("End commit instant for reading, the commit time format should be 'yyyyMMddHHmmss'");
 
+  public static final ConfigOption<Boolean> READ_SPEED_LIMIT_ENABLED = ConfigOptions
+      .key("read.speed.limit.enabled")
+      .booleanType()
+      .defaultValue(false)
+      .withDescription("Enable stream read speed limit, avoiding the risk of oom caused by the streamReadMonitor "
+          + " loading too many metadata at once");
+
+  public static final ConfigOption<Integer> READ_SPEED_LIMIT_COMMITS = ConfigOptions
+      .key("read.speed.limit.commits")
+      .intType()
+      .defaultValue(5)
+      .withDescription("The maximum number of commits allowed streamReadMonitor to read in each poll");
+
```

Comment:
`The maximum number of commits allowed streamReadMonitor to read in each poll` ->
`The maximum number of commits allowed to read in each instant check, if it is streaming read, the avg read instants number per-second would be 'read.commits.limit'/'read.streaming.check-interval', by default no limit`.